File: nodejs-service/src/rabbitMQ/factories/consumerFactory.ts
import container from "../../DI";
import { LogsConsumer, notiConsumer, UserConsumer } from "../consumers";
class ConsumerFactory {
  static createConsumer(queueName: string) {
    switch (queueName) {
      case "user_queue":
        return new UserConsumer();
      case "noti_queue":
        return new notiConsumer();
        case "logger_queue":
          return container.get<LogsConsumer>(LogsConsumer)
      default:
        throw new Error("Unknown queue type");
    }
  }
}

export default ConsumerFactory;

File: dotnet/backend-micro/backend-micro/RabbitMQ/Factories/ConsumerFactory.cs
using System;
using RabbitMQ.Client;
using RabbitMQ.Client.Events;
using System.Text;
using System.Threading.Tasks;
using backend_micro.DTO;
using backend_micro.RabbitMQ.Consumers;

namespace backend_micro.RabbitMQ.Factories
{
    public class ConsumerFactory
    {
        public static IConsumer CreateConsumer(string queueName)
        {
            return queueName switch
            {
                "user_queue" => new UserConsumer(),
                //"noti_queue" => new NotificationConsumer(),
                _ => throw new ArgumentException($"No consumer found for queue: {queueName}")
            };
        }
    }
}
